### ISSUE TITLE
[console] Add support for project configuration files

### ISSUE THREAD

**Issue by gergelypap** [hechoendrupal/DrupalConsole#2725]
Drupal Console should have support for per-project configuration, [similar to Drush](https://github.com/drush-ops/drush/blob/master/examples/example.drushrc.php). This could be handy for setting up things related to specific project, instead of using the global `~/.console` settings.

**Solution**

* Drupal console should read any `.drupalrc` file in the following places:
  * For composer projects: in the root folder (outside Drupal root `/web` directory)
  * For old-school Drupal projects: either in `sites/default` folder or the docroot (where core, modules, etc. folders live)
* It should allow YAML and JSON formats.
* It should override any global setting.

**Proposal for settings**

A `.drushrc` should allow settings of at least the following things:

* Base URL of the site (for console outputs like `drupal user:login:url <uid>`
* Boolean flag whether it is a Composer project (when using the `--composer` setting)
* Drupal root path (e.g. when the project is installed via Composer; _also `drupal` should be able to run in container-aware mode, when running outside the `web` directory, but this is another issue_!)
* Aliases
* Per-project chain files
* Directory of full-site and DB backup
* Environments and their settings (like in `~/.console/sites`)
* Basically any setting for all `drupal list` commands...

(See full featured [examples of drushrc.php](https://github.com/drush-ops/drush/blob/master/examples/example.drushrc.php))

What do you guys think?

**Comment by jmolivas** [comment]
@gergelypap This currently supported already and will be improved on RC-1.

### Actual places where DrupalConsole, search for configurations 
* Global user home `~/.console/`
* Drupal root path  `/path/to/site/console`

### Starting on RC-1
* DrupalConsole must be installed per site and the tool will let you know what to do if not currently installed.

![drupal-not-installed](https://cloud.githubusercontent.com/assets/366275/18703618/7e6354d0-7f9b-11e6-9fcc-5e2522f6f186.png)

* DrupalConsole will detect if no config file and recommend you to execute the init command

![drupal-no-config](https://cloud.githubusercontent.com/assets/366275/18703385/8c503b36-7f9a-11e6-8e7b-08e79ea60317.png)

* DrupalConsole will also guide you when creating the configuration file, and ask you to do this per site or globally if is local it will ask you for Drupal root path (`web`, `docroot`, `other`)

![drupal-init-1](https://cloud.githubusercontent.com/assets/366275/18703559/32a7c7c4-7f9b-11e6-8ad6-8f81f954d842.png)

![drupal-init-2](https://cloud.githubusercontent.com/assets/366275/18703560/32ab96e2-7f9b-11e6-8cd2-de0f5c6f1559.png)



* DrupalConsole

**Comment by jmolivas** [comment]
@gagarine  Already fixed on RC-1

#### Per site or global values for:
* Base URL of the site (for console outputs like drupal user:login:url <uid>
* Boolean flag whether it is a Composer project (when using the --composer setting)
* Drupal root path (e.g. when the project is installed via Composer; also drupal should be able to run in container-aware mode, when running outside the web directory, but this is another issue!)
* Aliases
* Per-project chain files at `/path/to/site/console/chain` 
* Directory of full-site and DB backup
* Environments and their settings (like in `~/.console/sites` at `/path/to/site/console` 
* Set any global application options like `--root`, `--composer`, `learning`, `etc`
* Set options and arguments per command.

This is an example of a config file on a site I am currently working on (I am using DC master branch not beta-5)
```
application:
    environment: prod
    language: en
    editor: vim
    temp: /tmp
    develop: 'false'
    command: about
    checked: 'true'
    remote:
        user: root
        port: '22'
        console: /usr/local/bin/drupal
        keys:
            public: ~/.ssh/id_rsa.pub
            private: ~/.ssh/id_rsa
            passphrase: ~/.ssh/passphrase.txt
    disable:
        modules:
          - metatag
        commands: {}
        namespaces: {}
    options:
        learning: false
        examples: false
        generate-inline: false
        generate-chain: false
        composer: false
        root: docroot
```

**Comment by jmolivas** [comment]
I also have some chains files we use to installl some modules and fill the site with dummy data after every rebuild.

`/path/to/site/console/chain/after-build.yml`
```
commands:
  - command: module:install
    options:
        root: docroot
        latest: true
    arguments:
        module:
            - features_ui
            - views_ui
  - command: create:terms
    options:
        root: docroot
        limit: 50
  - command: create:nodes
    options:
        limit: 50
        time-range: 604800
        root: docroot
    arguments:
        content-types:
            - blog_post
            - news_detail
            - video_detail
```

**Comment by gergelypap** [comment]
Wow... :open_mouth: Now this is great news, thanks for pointing it out. I have searched in the source, but found nothing.

Then the only thing we _really_ should improve on, is the project documentation :wink:

**Comment by jmolivas** [comment]
@gergelypap RC-4 was released and this si supported now.

Next step add to the docs.

**Comment by nvaken** [comment]
Not entirely sure what this means. I'm searching for a possibility to add `yml` project-specific for a project. So ideally, I would like to add targets into a file in `/path/to/composer/project-folder/console/`, is this possible? 

Additionally, I see the flag `--composer` mentioned to define if it's a composer project, I have not yet found where this flag is being defined. Could you elaborate? Thanks!

**Comment by jmolivas** [comment]
@nvaken when you mentioned `/path/to/composer/project-folder/console/` it does mean the path where drupal was downloaded ?

The `--composer`is no longer valid as global option.

**Comment by nvaken** [comment]
Hi @jmolivas, thanks for your quick reply. I meant a composer directory, like:

```
composer.json
web/[Drupal Install]
console/
```

A few ago I managed to get this working, it seems that is does indeed work, without configuration. I only needed to create a `console/sites` folder to add the `.yml` files to.
